scripts/install/opt/seagate/cortx/motr/libexec/motr_cfg.sh

```diff
@@ -160,22 +160,33 @@ set_key_value() # ARG1 [KEY] ARG2 [VALUE] ARG3 [FILE]
 do_m0provision_action()
 {
     local CFG_LINE=""
+    local LVM_SIZE_MIN=0
     local LVM_SIZE=0
 
-    msg "Configuring host [`hostname -f`]"
+    if [ ! -x /usr/sbin/lvs ]; then
+        err "lvs command not available."
+    fi
 
-    MD_DEVICE=$(lvs -o lv_path | grep "lv_raw_metadata" | head -1)
-    if [[ $? -eq 0 && $MD_DEVICE != "" ]];then
-        LVM_SIZE=$(lvs $MD_DEVICE  -o LV_SIZE \
-              --noheadings --units b --nosuffix | xargs)
+    msg "Configuring host [`hostname -f`]"
+    MD_DEVICES=($(lvs -o lv_path 2>/dev/null | grep "lv_raw_metadata" | grep srvnode  | sort -u))
 
+    for i in ${MD_DEVICES[@]};
+    do
+        LVM_SIZE=$(lvs $i -o LV_SIZE \
+                       --noheadings --units b --nosuffix | xargs)
         ANY_ERR=$(echo $LVM_SIZE | grep -i ERROR | wc -l)
-        if [[ ( "$ANY_ERR" != "0" ) || ( -z $LVM_SIZE ) ]]; then
-            err "lvs $MD_DEVICE command failed."
-            msg "[$LVM_SIZE]"
-        elif [[ $LVM_SIZE -ne 0 ]];then
-            sed -i "s/MOTR_M0D_IOS_BESEG_SIZE=.*/MOTR_M0D_IOS_BESEG_SIZE=$LVM_SIZE/g" $MOTR_CONF_FILE
-        fi
+        if [[ ( "$ANY_ERR" != "0" ) || ( -z "$LVM_SIZE" ) ]]; then
+            err "lvs $i command failed."
+            msg "[$LVM_SIZE_MIN]"
+	elif [[ "$LVM_SIZE" -lt "$LVM_SIZE_MIN" || "$LVM_SIZE_MIN" -eq 0 ]]; then
+	    LVM_SIZE_MIN=$LVM_SIZE
+	fi
+    done
+    if [ "$LVM_SIZE_MIN" -eq 0 ]; then
+	 err "lvs size invalid [$LVM_SIZE_MIN]"
```

Comment:
Shouldnt we return from here in case of error ?

Reply:
This check is to update IOS_BESEG_SIZE only as per metadata volume size, if it failed to get lvm size it will not update it. 
but same function further will check and update rest of the config parameter, so can't return from this place.